shenyu-web/src/main/java/org/apache/shenyu/web/filter/CrossFilter.java

```diff
@@ -58,15 +60,14 @@ public Mono<Void> filter(@NonNull final ServerWebExchange exchange, @NonNull fin
         if (CorsUtils.isCorsRequest(request)) {
             ServerHttpResponse response = exchange.getResponse();
             HttpHeaders headers = response.getHeaders();
-            // "Access-Control-Allow-Origin"
+            // finger allow origins
             final String origin = request.getHeaders().getOrigin();
-            if (this.filterConfig.isAllowedAnyOrigin()) {
-                headers.set(HttpHeaders.ACCESS_CONTROL_ALLOW_ORIGIN, origin);
-            } else if (Objects.nonNull(this.filterConfig.getAllowedOrigin())
-                    && CollectionUtils.isNotEmpty(this.filterConfig.getAllowedOrigin().getPrefixes())) {
+            boolean allowCors = this.filterConfig.isAllowedAnyOrigin();
+            if (!allowCors && Objects.nonNull(this.filterConfig.getAllowedOrigin())) {
                 final String scheme = exchange.getRequest().getURI().getScheme();
                 final CrossFilterConfig.AllowedOriginConfig allowedOriginConfig = this.filterConfig.getAllowedOrigin();
-                Set<String> allowedOrigin = allowedOriginConfig.getPrefixes()
+                Set<String> allowedOrigin = new HashSet<>();
```

Comment:
Set<String> allowedOrigin =  Optional.ofNullable(allowedOriginConfig.getPrefixes()).orElse(Collections.emptySet())...

I think this will be abit moe concise.